ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -213,6 +216,32 @@ inline uint32_t get_per_core_factor(const Tensor& input_tensor_a, const Tensor&
     return 1;
 }
 
+inline std::vector<uint32_t> get_mutlti_dim_per_core_factor(const Tensor& input_tensor_a, const Tensor& input_tensor_b, uint32_t per_core_M, uint32_t per_core_N, uint32_t in0_block_w, uint32_t interm_cb_size) {
+    uint32_t max_l1_space = get_max_l1_space(input_tensor_a);
```

Comment:
just curious why this takes into account only one tensor? and not second tensor as well?

Reply:
it's just for getting the device obj from the tensor, both tensor should have same device